### PR TITLE
Share status

### LINKED ISSUE #20: [💡 Feature request] create a copy to clipboard with the current informations on the page
One cool way to use this webapp would be to send it to non techincal users to get a report of their situation (useful for debugging)

The easiest way would be to include a copy to clipboard button that will copy the information in a dev digestable way. The step up could be add the ability to add a query parameter (reportTo) and onload send an email to it with the current information (so that one could just send a link with their email address setupped and immediately get back a feedback.

Obviosuly this second one it's far more expensive and it also has possible security implications to reason of but it could even be a form of profit for you (it could very easily become a paid feature but i don't know if that's where this project is headed).

If you are ok with those i can even start to work on those (to be fair i never used nuxt and i barely used vue but it shuldn't be to difficult and that is what code reviews are for right? RIGHT?)

Joking aside let me know what do you think about the proposal and let me know if you are ok with me working on them.

### PR DESCRIPTION
**Pull request by toniengelhardt**
Generated from #27. Closes #20.

- Simplifies status logic
- Tweaks
- Style

### CODE REVIEW

**Comment by paoloricciuti** [comment]
Ok i've added base 64 encode and decode. I've also fixed a bug in the decode status.

Now the only problem is...i can't push to this branch lol

Would you prefer that i merge this into my fork and we go back to the other PR?

**Comment by paoloricciuti** [comment]
Also testing is failing because it can't find webApisData...importing it manually works but than it tryes to import "#components" and still fail. I'm trying to work around this without much success

**Comment by toniengelhardt** [comment]
Excellent! 

Yes, let's go back to your PR. Just created the second one bc I also didn't understand how to easily add on top of yours without checking out your fork separately. 

Don't worry, just merge your stuff and I'll take a look at the test.

**Comment by paoloricciuti** [comment]
Perfect, pushed on the other PR.

**Comment by paoloricciuti** [comment]
@toniengelhardt it seems like it's prerendering index.html...i'm trying to force ssr to see if it fixes it

File: pages/index.vue [hunk 1]
```diff
@@ -85,9 +85,14 @@ const totalAPICount = computed(() => webApiList.value.length)
...
 }
-useTestWebApis()
 
-watch(() => displayMode.value, (newVal, oldVal) => {
-  console.log(`DisplayMode changed from '${oldVal}' to '${newVal}'`)
+// watch(() => displayMode.value, (newVal, oldVal) => {
+//   console.log(`DisplayMode changed from '${oldVal}' to '${newVal}'`)
+// })
+
+onMounted(() => {
```
Comment (paoloricciuti) [inline pages/index.vue:93]: Running this only onMounted will leave us without SSR tho

Comment (paoloricciuti) [reply pages/index.vue:93]: With this https://github.com/toniengelhardt/webapicheck/pull/28#issuecomment-1438582395 we can fix the prerendering issue and get SSR when there's a query param and populating the fields on the client when there's not.

**Comment by toniengelhardt** [comment]
Hey @paoloricciuti, I just merged the PR in prod. 

Some notes:
- Linting and tests are working now, but only in the CI, locally I somehow get errors with sitemap.xml (probably related to the nuxt-seo-kit), really strange. I switched the setup from vitest to [nuxt-vitest](https://github.com/danielroe/nuxt-vitest), I think that was the issue.
- I switched the logic back to always `onMounted`. Reasons are the following: it feels off that the `onMounted` hook is in the composable (bad readability) and it will probably cause problems in the future if the logic is not consistent, bc we will always have to deal with 2 cases instead of just one. Since the performance gain is minimal I think this is the better solution and keeps things simple. This also has the advantage that we don't need to distinguish between `btoa` and `Buffer` which will probably also cause issues if the environment is suddenly not node or browser. This way we can just always use `window.btoa`.
- I switched the encoding/decoding from `parseInt` to `BigInt` because there were rounding errors.

For future reference, the changelog is auto-generated, so if you want your commits to be included you need to follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) guidelines (see contribution section in README), like `feat: share button` for example.

All in all, EXCELLENT WORK, I think the feature is really cool! 🙏🏽

**Comment by paoloricciuti** [comment]
> Hey @paoloricciuti, I just merged the PR in prod.
> 
> Some notes:
> 
> * Linting and tests are working now, but only in the CI, locally I somehow get errors with sitemap.xml (probably related to the nuxt-seo-kit), really strange. I switched the setup from vitest to [nuxt-vitest](https://github.com/danielroe/nuxt-vitest), I think that was the issue.
> * I switched the logic back to always `onMounted`. Reasons are the following: it feels off that the `onMounted` hook is in the composable (bad readability) and it will probably cause problems in the future if the logic is not consistent, bc we will always have to deal with 2 cases instead of just one. Since the performance gain is minimal I think this is the better solution and keeps things simple. This also has the advantage that we don't need to distinguish between `btoa` and `Buffer` which will probably also cause issues if the environment is suddenly not node or browser. This way we can just always use `window.btoa`.
> * I switched the encoding/decoding from `parseInt` to `BigInt` because there were rounding errors.
> 
> For future reference, the changelog is auto-generated, so if you want your commits to be included you need to follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) guidelines (see contribution section in README), like `feat: share button` for example.
> 
> All in all, EXCELLENT WORK, I think the feature is really cool! 🙏🏽

All right man, sorry for the messy code, first time working with nuxt. If you feel it's better to leave the logic in the onMounted you have all the right to do so 😄 

For the future i'll try to follow the conventional commit rules (it's an habit i have to start following).

Happy to have contributed to your project and thanks for the patience 🙇

**Comment by toniengelhardt** [comment]
No worries, I'm actually really impressed what you came up with given you have never worked with Nuxt!

Thanks a lot 🙏🏽 🙏🏽 🙏🏽

**Comment by paoloricciuti** [comment]
Thank you for your project 💪

**Pull request merged**